datadog/resource_datadog_dashboard.go

```diff
@@ -4344,6 +4361,183 @@ func buildTerraformServiceLevelObjectiveDefinition(datadogDefinition datadogV1.S
 	return terraformDefinition
 }
 
+//
+// Geomap Widget Definition helpers
+//
+func getGeomapDefinitionSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"request": {
+			Description: "Nested block describing the request to use when displaying the widget. Multiple `request` blocks are allowed with the structure below (exactly one of `q`, `log_query` or `rum_query` is required within the `request` block).",
+			Type:        schema.TypeList,
+			Optional:    true,
+			Elem: &schema.Resource{
+				Schema: getGeomapRequestSchema(),
+			},
+		},
+		"style": {
+			Description: "Style of the widget graph. One nested block is allowed with the structure below.",
+			Type:        schema.TypeList,
+			Optional:    true,
+			MaxItems:    1,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"palette": {
+						Description: "The color palette to apply to the widget.",
+						Type:        schema.TypeString,
+						Required:    true,
+					},
+					"palette_flip": {
+						Description: "Boolean indicating whether to flip the palette tones.",
+						Type:        schema.TypeBool,
+						Required:    true,
+					},
+				},
+			},
+		},
+		"view": {
+			Description: "The view of the world that the map should render.",
+			Type:        schema.TypeList,
+			Required:    true,
+			MaxItems:    1,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"focus": {
+						Description: "The 2-letter ISO code of a country to focus the map on. Or `WORLD`.",
+						Type:        schema.TypeString,
+						Required:    true,
+					},
+				},
+			},
+		},
+		"title": {
+			Description: "The title of the widget.",
+			Type:        schema.TypeString,
+			Optional:    true,
+		},
+		"title_size": {
+			Description: "The size of the widget's title. Default is 16.",
+			Type:        schema.TypeString,
+			Optional:    true,
+		},
+		"title_align": {
+			Description:  "The alignment of the widget's title. One of `left`, `center`, or `right`.",
+			Type:         schema.TypeString,
+			ValidateFunc: validators.ValidateEnumValue(datadogV1.NewWidgetTextAlignFromValue),
+			Optional:     true,
+		},
+		"time":      getDeprecatedTimeSchema(),
```

Comment:
I think we should remove this line. As I understand, `live_span` covers the same functionality, but without the TypeMap deprecated path.

Reply:
Done in commit [82940d4](https://github.com/DataDog/terraform-provider-datadog/pull/954/commits/82940d4ce6e86936c65dd22e85b98ba68bcc21ef).